src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+package org.fundacionjala.virtualassistant.player.spotify.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.servlet.view.RedirectView;
+
+import java.io.IOException;
+
+@Component
+public class SpotifyClient implements MusicClient {
+
+    @Value("${spotify.client.id}")
+    private String clientId;
+
+    @Value("${spotify.client.secret}")
+    private String clientSecret;
+
+    @Value("${spotify.redirect.uri}")
+    private String redirectUri;
+
+    private String accessToken;
+
+    @Override
+    public boolean playSongOnDevice(String trackUri) {
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            headers.set("Authorization", "Bearer " + accessToken);
+            headers.setContentType(MediaType.APPLICATION_JSON);
+
+            String requestBody = "{\"uris\": [\"" + trackUri + "\"]}";
+            HttpEntity<String> entity = new HttpEntity<>(requestBody, headers);
+
+            RestTemplate restTemplate = new RestTemplate();
+            ResponseEntity<String> response = restTemplate.exchange(
+                    "https://api.spotify.com/v1/me/player/play",
+                    HttpMethod.PUT,
+                    entity,
+                    String.class
+            );
+
+            return response.getStatusCode() == HttpStatus.NO_CONTENT;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void setAccessToken(String accessToken) {
+        this.accessToken = accessToken;
+    }
+
+    public RedirectView redirectToSpotifyAuthorization() {
+        String spotifyAuthUrl = "https://accounts.spotify.com/authorize?client_id=" + clientId +
+                "&response_type=code&redirect_uri=" + redirectUri + "&scope=user-read-private%20user-read-email user-library-read user-follow-read user-read-playback-state app-remote-control user-modify-playback-state";
+        return new RedirectView(spotifyAuthUrl);
+    }
+
+    public ResponseEntity<String> spotifyCallback(String code) {
+        accessToken = exchangeAuthCodeForAccessToken(code);
+        return ResponseEntity.ok("Logged in successfully");
+    }
+
+    @Override
+    public String exchangeAuthCodeForAccessToken(String authorizationCode) {
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
Dont create a new instance of RestTemplate for each call, define this variable as class variables